### PR TITLE
dotCMS/core#11896 fix VelVarName generator for Categories

### PR DESCRIPTION
**Pull request by joseorsini**
Tested on Master + H2 database. 

Works as expected.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategorySQL.java [hunk 1]
```diff
@@ -40,5 +40,9 @@ public String getSortedChildren() {
 				"category.inode = tree.child and tree.parent = ? and category_1_.inode = category.inode " +
 				" and category_1_.type = 'category'";
 	}
+	
+	public String getVelocityVarNameCount() {
+	    return "select count(*) as test from category where category_velocity_var_name like ?";
+	}
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategorySQL.java]: Why a method?, I think is better:
public static final VELOCITY_VAR_NAME_COUNT_SQL = "select count(*) as test from category where category_velocity_var_name like ?"

File: dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPIImpl.java [hunk 2]
```diff
@@ -565,8 +566,12 @@ public boolean isParent(Category givenChild, Category givenParent, User user) {
 		return false;
 	}
 
-
-
-
+	public synchronized String suggestVelocityVarName(final String categoryVelVarName) throws DotDataException {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPIImpl.java]: why is this synchronized?

File: dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategorySQL.java (hunk 1 above)
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategorySQL.java]: I think they use a method in order to be override by a subclass, such as OracleCategorySQL
However the constants still valid in this context, the method will return the constant.

Comment (joseorsini) [inline dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategorySQL.java]: What @jdotcms said. There's a chance that the whole CategoryFactory implementation can see a refactoring in the same way we did for content types (mostly for SQL operations), and current logic here (which it's the same for fields and content types) was missing for categories, which also require a VelVarName once they're created

File: dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPIImpl.java (hunk 2 above)
Comment (joseorsini) [inline dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPIImpl.java]: as in my comment before, it's based on the same code we have for ContentTypeAPI that generates the VelocityVarName for Content Types. Categories had this loose end.

**Review by dsilvam — APPROVED** [review]

**Pull request merged**